{"metadata": {"language_info": {"name": "bash", "codemirror_mode": "shell", "mimetype": "text/x-sh", "file_extension": ".sh"}, "kernelspec": {"name": "bash", "display_name": "Bash", "language": "bash"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "markdown", "source": "This bash script runs Popcorn, a program for estimating the correlation of causal variant effect, on the Pre-Confluence Project Data.", "metadata": {"tags": []}}, {"cell_type": "code", "source": "dx cd\ndx cd Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/corr_files\ndx ls", "metadata": {"trusted": true, "tags": []}, "execution_count": 1, "outputs": [{"name": "stdout", "text": "\u001b[1m\u001b[34m1KG_AFR_LD_corr_files/\u001b[0m\nAFR_AMR_Overall_HM3.txt\nAFR_AMR_Overall_HM3.txt.e.log\nAFR_AMR_Overall_HM3.txt.o.log\nAFR_AMR_Overall_HM3_01.txt\nAFR_AMR_Overall_HM3_01.txt.e.log\nAFR_AMR_Overall_HM3_01.txt.o.log\nAFR_AMR_Overall_HM3_05.txt\nAFR_AMR_Overall_HM3_05.txt.e.log\nAFR_AMR_Overall_HM3_05.txt.o.log\nAFR_AMR_Overall_HM3_MEGA.txt\nAFR_AMR_Overall_HM3_MEGA.txt.e.log\nAFR_AMR_Overall_HM3_MEGA.txt.o.log\nAFR_AMR_Overall_HM3_MEGA_01.txt\nAFR_AMR_Overall_HM3_MEGA_01.txt.e.log\nAFR_AMR_Overall_HM3_MEGA_01.txt.o.log\nAFR_AMR_Overall_HM3_MEGA_05.txt\nAFR_AMR_Overall_HM3_MEGA_05.txt.e.log\nAFR_AMR_Overall_HM3_MEGA_05.txt.o.log\nAFR_EAS_Overall_HM3_01.txt\nAFR_EAS_Overall_HM3_01.txt.e.log\nAFR_EAS_Overall_HM3_01.txt.o.log\nAFR_EAS_Overall_HM3_05.txt\nAFR_EAS_Overall_HM3_05.txt.e.log\nAFR_EAS_Overall_HM3_05.txt.o.log\nAFR_EAS_Overall_HM3_MEGA_01.txt\nAFR_EAS_Overall_HM3_MEGA_01.txt.e.log\nAFR_EAS_Overall_HM3_MEGA_01.txt.o.log\nAFR_EAS_Overall_HM3_MEGA_05.txt\nAFR_EAS_Overall_HM3_MEGA_05.txt.e.log\nAFR_EAS_Overall_HM3_MEGA_05.txt.o.log\nAFR_EUR_Overall_HM3_01.txt\nAFR_EUR_Overall_HM3_01.txt.e.log\nAFR_EUR_Overall_HM3_01.txt.o.log\nAFR_EUR_Overall_HM3_05.txt\nAFR_EUR_Overall_HM3_05.txt.e.log\nAFR_EUR_Overall_HM3_05.txt.o.log\nAFR_EUR_Overall_HM3_MEGA_01.txt\nAFR_EUR_Overall_HM3_MEGA_01.txt.e.log\nAFR_EUR_Overall_HM3_MEGA_01.txt.o.log\nAFR_EUR_Overall_HM3_MEGA_05.txt\nAFR_EUR_Overall_HM3_MEGA_05.txt.e.log\nAFR_EUR_Overall_HM3_MEGA_05.txt.o.log\nAMR_EAS_Overall_HM3.txt\nAMR_EAS_Overall_HM3.txt.e.log\nAMR_EAS_Overall_HM3.txt.o.log\nAMR_EAS_Overall_HM3_01.txt\nAMR_EAS_Overall_HM3_01.txt.e.log\nAMR_EAS_Overall_HM3_01.txt.o.log\nAMR_EAS_Overall_HM3_05.txt\nAMR_EAS_Overall_HM3_05.txt.e.log\nAMR_EAS_Overall_HM3_05.txt.o.log\nAMR_EAS_Overall_HM3_MEGA.txt\nAMR_EAS_Overall_HM3_MEGA.txt.e.log\nAMR_EAS_Overall_HM3_MEGA.txt.o.log\nAMR_EAS_Overall_HM3_MEGA_01.txt\nAMR_EAS_Overall_HM3_MEGA_01.txt.e.log\nAMR_EAS_Overall_HM3_MEGA_01.txt.o.log\nAMR_EAS_Overall_HM3_MEGA_05.txt\nAMR_EAS_Overall_HM3_MEGA_05.txt.e.log\nAMR_EAS_Overall_HM3_MEGA_05.txt.o.log\nAMR_EUR_Overall_HM3.txt\nAMR_EUR_Overall_HM3.txt.e.log\nAMR_EUR_Overall_HM3.txt.o.log\nAMR_EUR_Overall_HM3_01.txt\nAMR_EUR_Overall_HM3_01.txt.e.log\nAMR_EUR_Overall_HM3_01.txt.o.log\nAMR_EUR_Overall_HM3_05.txt\nAMR_EUR_Overall_HM3_05.txt.e.log\nAMR_EUR_Overall_HM3_05.txt.o.log\nAMR_EUR_Overall_HM3_MEGA.txt\nAMR_EUR_Overall_HM3_MEGA.txt.e.log\nAMR_EUR_Overall_HM3_MEGA.txt.o.log\nAMR_EUR_Overall_HM3_MEGA_01.txt\nAMR_EUR_Overall_HM3_MEGA_01.txt.e.log\nAMR_EUR_Overall_HM3_MEGA_01.txt.o.log\nAMR_EUR_Overall_HM3_MEGA_05.txt\nAMR_EUR_Overall_HM3_MEGA_05.txt.e.log\nAMR_EUR_Overall_HM3_MEGA_05.txt.o.log\nEAS_EUR_Overall_HM3_01.txt\nEAS_EUR_Overall_HM3_01.txt.e.log\nEAS_EUR_Overall_HM3_01.txt.o.log\nEAS_EUR_Overall_HM3_05.txt\nEAS_EUR_Overall_HM3_05.txt.e.log\nEAS_EUR_Overall_HM3_05.txt.o.log\nEAS_EUR_Overall_HM3_MEGA_01.txt\nEAS_EUR_Overall_HM3_MEGA_01.txt.e.log\nEAS_EUR_Overall_HM3_MEGA_01.txt.o.log\nEAS_EUR_Overall_HM3_MEGA_05.txt\nEAS_EUR_Overall_HM3_MEGA_05.txt.e.log\nEAS_EUR_Overall_HM3_MEGA_05.txt.o.log\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "# Install Popcorn", "metadata": {}}, {"cell_type": "code", "source": "python --version    # check running python3", "metadata": {"trusted": true, "tags": []}, "execution_count": 2, "outputs": [{"name": "stdout", "text": "Python 3.9.16\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "git clone https://github.com/brielin/Popcorn.git\ncd Popcorn\npip install .", "metadata": {"trusted": true, "tags": []}, "execution_count": 3, "outputs": [{"name": "stdout", "text": "Cloning into 'Popcorn'...\nremote: Enumerating objects: 639, done.        \nremote: Counting objects: 100% (83/83), done.        \nremote: Compressing objects: 100% (56/56), done.        \nremote: Total 639 (delta 43), reused 61 (delta 26), pack-reused 556        \nReceiving objects: 100% (639/639), 19.28 MiB | 27.16 MiB/s, done.\nResolving deltas: 100% (392/392), done.\n\u001b[33mWARNING: The directory '/home/dnanexus/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\u001b[0m\u001b[33m\n\u001b[0mProcessing /opt/notebooks/Popcorn\n  Preparing metadata (setup.py) ... \u001b[?25ldone\n\u001b[?25hRequirement already satisfied: numpy in /opt/conda/lib/python3.9/site-packages (from popcorn==1.0.0) (1.23.5)\nRequirement already satisfied: scipy in /opt/conda/lib/python3.9/site-packages (from popcorn==1.0.0) (1.9.3)\nRequirement already satisfied: pandas in /opt/conda/lib/python3.9/site-packages (from popcorn==1.0.0) (1.5.3)\nCollecting pysnptools\n  Downloading pysnptools-0.5.13.tar.gz (223 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m223.2/223.2 kB\u001b[0m \u001b[31m108.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25h  Installing build dependencies ... \u001b[?25ldone\n\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n\u001b[?25hCollecting bottleneck\n  Downloading Bottleneck-1.4.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (355 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m355.1/355.1 kB\u001b[0m \u001b[31m254.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting statsmodels\n  Downloading statsmodels-0.14.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (10.8 MB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m10.8/10.8 MB\u001b[0m \u001b[31m238.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n\u001b[?25hRequirement already satisfied: python-dateutil>=2.8.1 in /opt/conda/lib/python3.9/site-packages (from pandas->popcorn==1.0.0) (2.8.2)\nRequirement already satisfied: pytz>=2020.1 in /opt/conda/lib/python3.9/site-packages (from pandas->popcorn==1.0.0) (2023.3.post1)\nCollecting more-itertools>=8.7.0\n  Downloading more_itertools-10.3.0-py3-none-any.whl (59 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m59.2/59.2 kB\u001b[0m \u001b[31m181.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting cloudpickle>=2.2.0\n  Downloading cloudpickle-3.0.0-py3-none-any.whl (20 kB)\nRequirement already satisfied: psutil>=5.6.3 in /opt/conda/lib/python3.9/site-packages (from pysnptools->popcorn==1.0.0) (5.9.6)\nCollecting h5py>=2.10.0\n  Downloading h5py-3.11.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (5.3 MB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m5.3/5.3 MB\u001b[0m \u001b[31m318.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting bed-reader[samples]>=1.0.5\n  Downloading bed_reader-1.0.5-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (6.3 MB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m6.3/6.3 MB\u001b[0m \u001b[31m267.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hRequirement already satisfied: packaging>=21.3 in /opt/conda/lib/python3.9/site-packages (from statsmodels->popcorn==1.0.0) (23.0)\nCollecting patsy>=0.5.6\n  Downloading patsy-0.5.6-py2.py3-none-any.whl (233 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m233.9/233.9 kB\u001b[0m \u001b[31m205.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting pooch>=1.5.0\n  Downloading pooch-1.8.2-py3-none-any.whl (64 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m64.6/64.6 kB\u001b[0m \u001b[31m174.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hRequirement already satisfied: six in /opt/conda/lib/python3.9/site-packages (from patsy>=0.5.6->statsmodels->popcorn==1.0.0) (1.16.0)\nRequirement already satisfied: requests>=2.19.0 in /opt/conda/lib/python3.9/site-packages (from pooch>=1.5.0->bed-reader[samples]>=1.0.5->pysnptools->popcorn==1.0.0) (2.28.2)\nRequirement already satisfied: platformdirs>=2.5.0 in /opt/conda/lib/python3.9/site-packages (from pooch>=1.5.0->bed-reader[samples]>=1.0.5->pysnptools->popcorn==1.0.0) (4.0.0)\nRequirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.9/site-packages (from requests>=2.19.0->pooch>=1.5.0->bed-reader[samples]>=1.0.5->pysnptools->popcorn==1.0.0) (2023.11.17)\nRequirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.9/site-packages (from requests>=2.19.0->pooch>=1.5.0->bed-reader[samples]>=1.0.5->pysnptools->popcorn==1.0.0) (2.0.4)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.9/site-packages (from requests>=2.19.0->pooch>=1.5.0->bed-reader[samples]>=1.0.5->pysnptools->popcorn==1.0.0) (3.4)\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.9/site-packages (from requests>=2.19.0->pooch>=1.5.0->bed-reader[samples]>=1.0.5->pysnptools->popcorn==1.0.0) (1.26.15)\nBuilding wheels for collected packages: popcorn, pysnptools\n  Building wheel for popcorn (setup.py) ... \u001b[?25ldone\n\u001b[?25h  Created wheel for popcorn: filename=popcorn-1.0.0-py3-none-any.whl size=17228 sha256=455cab6b0b3ce46ab35eb863f0e83bbf89be55b00651a41657413843c8dbeabb\n  Stored in directory: /tmp/pip-ephem-wheel-cache-5xlad8um/wheels/a9/9f/13/e342c997ac040339b4db0e12f9dffa62e71c2f311d26a60145\n  Building wheel for pysnptools (pyproject.toml) ... \u001b[?25ldone\n\u001b[?25h  Created wheel for pysnptools: filename=pysnptools-0.5.13-py3-none-any.whl size=280297 sha256=a04c74f79604c3a28ec1b54bcc415c30a540fa9abb1ba0b547d9f67a4bf6f1c5\n  Stored in directory: /tmp/pip-ephem-wheel-cache-5xlad8um/wheels/75/82/1a/18dffc4d0aba3fcae3f8cd05f05c9509a3701550745c024530\nSuccessfully built popcorn pysnptools\nInstalling collected packages: patsy, more-itertools, h5py, cloudpickle, bottleneck, bed-reader, pooch, statsmodels, pysnptools, popcorn\nSuccessfully installed bed-reader-1.0.5 bottleneck-1.4.0 cloudpickle-3.0.0 h5py-3.11.0 more-itertools-10.3.0 patsy-0.5.6 pooch-1.8.2 popcorn-1.0.0 pysnptools-0.5.13 statsmodels-0.14.2\n\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n\u001b[0m\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "popcorn -h   # check popcorn was installed correctly", "metadata": {"trusted": true, "tags": []}, "execution_count": 4, "outputs": [{"name": "stdout", "text": "usage: popcorn [-h] {compute,fit} ...\n\npositional arguments:\n  {compute,fit}  Program mode: \"compute\" for computing covariance scores,\n                 \"fit\" for fitting summary statistics with pre-computed\n                 covariance scores.\n\noptional arguments:\n  -h, --help     show this help message and exit\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "# Run Popcorn", "metadata": {}}, {"cell_type": "code", "source": "popcorn fit -h    # shows fit usage", "metadata": {"trusted": true, "tags": []}, "execution_count": 5, "outputs": [{"name": "stdout", "text": "usage: popcorn fit [-h] [-v {0,1,2,3}] [--maf MAF] [--gen_effect] [--h2weight]\n                   [--from_bp FROM_BP] [--to_bp TO_BP] [--sfile SFILE]\n                   [--sfile1 SFILE1] [--sfile2 SFILE2] [--cfile CFILE]\n                   [--Ns NS] [--tol TOL] [--no_jackknife] [--K1 K1] [--P1 P1]\n                   [--K2 K2] [--P2 P2] [--no_intercept] [--plot_likelihood]\n                   [--use_mle]\n                   out\n\npositional arguments:\n  out                Specify output file.\n\noptional arguments:\n  -h, --help         show this help message and exit\n  -v {0,1,2,3}       Specify output verbosity.\n  --maf MAF          Specify MAF cutoff. We suggest at least 0.01.\n  --gen_effect       Compute genetic effect correlation instead of genetic\n                     impact correlation. This must be specified both when\n                     computing scores and fitting the model.\n  --h2weight         Incorporate per-SNP h2 weighting from file. The filename\n                     should be (radix of binary plink file).h2weight\n  --from_bp FROM_BP  Specify base position to start analysis.\n  --to_bp TO_BP      Specify base position to finish analysus.\n  --sfile SFILE      Specify tab delimited summary statistics file\n  --sfile1 SFILE1    Specify tab delimited summary statistics file for\n                     population 1\n  --sfile2 SFILE2    Specify tab delimited summary statistics file for\n                     population 2\n  --cfile CFILE      Specify covariance scores file\n  --Ns NS            Either 1) the number of overlapping samples or 2)\n                     proportion of overlapping samples or a 3) text file\n                     containing the number of overlapping samples per SNP ID.\n  --tol TOL          Specify the convergence tolerance for MLE. Unless you\n                     have convergence issues use default\n  --no_jackknife     Just compute the point estimate without computing the\n                     jackknife standard error.\n  --K1 K1            Specify population prevelance of binary phenotype 1.\n  --P1 P1            Specify study prevelance of binary phenotype 1.\n  --K2 K2            Specify population prevelance of binary phenotype 2.\n  --P2 P2            Specify study prevelance of binary phenotype 2.\n  --no_intercept     Fixes intercept of heritability to be 1.0.\n  --plot_likelihood  Plot the likehood as a function of the parameter.\n  --use_mle          Use original MLE estimator instead of more stable but\n                     higher variance regression estimator.\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "SCORES_PREF=/mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files\nSUMSTATS_PREF=/mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files", "metadata": {"trusted": true, "tags": []}, "execution_count": 6, "outputs": []}, {"cell_type": "code", "source": "dx ls ../scores_files", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": [{"name": "stdout", "text": "\u001b[1m\u001b[34m1KG_AFR_LD_scores_files/\u001b[0m\nAA_AMR_cleaned_scores.txt\nAA_EAS_cleaned_scores.txt\nAA_EUR_cleaned_scores.txt\nAFR_AMR_cleaned_scores.txt\nAFR_EAS_cleaned_scores.txt\nAFR_EUR_cleaned_scores.txt\nAMR_EAS_cleaned_scores.txt\nAMR_EAS_cleaned_scores.txt.e.log\nAMR_EAS_cleaned_scores.txt.o.log\nAMR_EUR_cleaned_scores.txt\nAMR_EUR_cleaned_scores.txt.e.log\nAMR_EUR_cleaned_scores.txt.o.log\nEAS_EUR_cleaned_scores.txt\nEAS_EUR_cleaned_scores.txt.e.log\nEAS_EUR_cleaned_scores.txt.o.log\nEAS_EUR_scores.txt\nEAS_EUR_scores.txt.e.log\nEAS_EUR_scores.txt.o.log\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# SUBSET=\"HM3_MEGA\"\nSUBSET=\"HM3\"\nPOPS=\"AFR AMR EAS EUR\"\n\nread -r -a POPSARR <<< \"$POPS\"\nNUMPOPS=${#POPSARR[@]}\n\nfor ((i = 0; i < NUMPOPS; i++))\ndo\n    POP1=${POPSARR[i]}\n    MAF1=\"01\"\n    if [ \"$POP1\" = \"AMR\" ]\n    then\n        MAF1=\"05\"\n    fi\n    \n    for ((j = i + 1; j < NUMPOPS; j++))\n    do\n        POP2=${POPSARR[j]}\n        MAF2=\"01\"\n        if [ \"$POP2\" = \"AMR\" ]\n        then\n            MAF2=\"05\"\n        fi\n\n        if [ \"$POP1\" != \"AMR\" ] && [ \"$POP2\" != \"AMR\" ]\n        then\n            break\n        fi\n\n        popcorn fit -v 1 \\\n           --sfile1 $SUMSTATS_PREF/${POP1}_Overall_${SUBSET}_${MAF1}.sumstats.txt \\\n           --sfile2 $SUMSTATS_PREF/${POP2}_Overall_${SUBSET}_${MAF2}.sumstats.txt \\\n           --cfile $SCORES_PREF/${POP1}_${POP2}_cleaned_scores.txt \\\n           ${POP1}_${POP2}_Overall_${SUBSET}.txt\n        \n        dx upload ${POP1}_${POP2}_Overall_${SUBSET}.txt*\n    done\ndone", "metadata": {"trusted": true, "tags": []}, "execution_count": 23, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 272 of 272 bytes (100%) AFR_AMR_Overall_HM3.txt\nID                          file-GpV9F0j4BzFp9P0Pq017Q8Pv\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_Overall_HM3.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu Jul 25 18:12:19 2024\nCreated by                  omjahagirdar\n via the job                job-GpV8Fyj4BzFVb8qgpzJFJ3b6\nLast modified               Thu Jul 25 18:12:19 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GpV9F104BzFpQ67J4412KBY3\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_Overall_HM3.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu Jul 25 18:12:20 2024\nCreated by                  omjahagirdar\n via the job                job-GpV8Fyj4BzFVb8qgpzJFJ3b6\nLast modified               Thu Jul 25 18:12:20 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,591 of 5,591 bytes (100%) AFR_AMR_Overall_HM3.txt.o.log\nID                          file-GpV9F184BzFp9P0Pq017Q8Q4\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_Overall_HM3.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu Jul 25 18:12:21 2024\nCreated by                  omjahagirdar\n via the job                job-GpV8Fyj4BzFVb8qgpzJFJ3b6\nLast modified               Thu Jul 25 18:12:21 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 268 of 268 bytes (100%) AMR_EAS_Overall_HM3.txt\nID                          file-GpV9F1Q4BzFp9P0Pq017Q8QG\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EAS_Overall_HM3.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu Jul 25 18:12:22 2024\nCreated by                  omjahagirdar\n via the job                job-GpV8Fyj4BzFVb8qgpzJFJ3b6\nLast modified               Thu Jul 25 18:12:23 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GpV9F1j4BzFyf6B87f22FFZF\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EAS_Overall_HM3.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu Jul 25 18:12:23 2024\nCreated by                  omjahagirdar\n via the job                job-GpV8Fyj4BzFVb8qgpzJFJ3b6\nLast modified               Thu Jul 25 18:12:24 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,590 of 5,590 bytes (100%) AMR_EAS_Overall_HM3.txt.o.log\nID                          file-GpV9F204BzFk1b382GK7469x\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EAS_Overall_HM3.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu Jul 25 18:12:24 2024\nCreated by                  omjahagirdar\n via the job                job-GpV8Fyj4BzFVb8qgpzJFJ3b6\nLast modified               Thu Jul 25 18:12:25 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 273 of 273 bytes (100%) AMR_EUR_Overall_HM3.txt\nID                          file-GpV9F284BzFqq3bXKz85K1qv\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EUR_Overall_HM3.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu Jul 25 18:12:25 2024\nCreated by                  omjahagirdar\n via the job                job-GpV8Fyj4BzFVb8qgpzJFJ3b6\nLast modified               Thu Jul 25 18:12:26 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GpV9F2Q4BzFVKYxqJ0K7v2ZV\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EUR_Overall_HM3.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu Jul 25 18:12:26 2024\nCreated by                  omjahagirdar\n via the job                job-GpV8Fyj4BzFVb8qgpzJFJ3b6\nLast modified               Thu Jul 25 18:12:27 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,590 of 5,590 bytes (100%) AMR_EUR_Overall_HM3.txt.o.log\nID                          file-GpV9F2j4BzFVKYxqJ0K7v2Zb\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EUR_Overall_HM3.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu Jul 25 18:12:27 2024\nCreated by                  omjahagirdar\n via the job                job-GpV8Fyj4BzFVb8qgpzJFJ3b6\nLast modified               Thu Jul 25 18:12:28 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# SUBSET=\"HM3_MEGA\"\nSUBSET=\"HM3\"\nMAF=\"05\"\nPOPS=\"AFR AMR EAS EUR\"\n# POPS=\"AFR EUR\"\n\nread -r -a POPSARR <<< \"$POPS\"\nNUMPOPS=${#POPSARR[@]}\n\n# for ((i = 0; i < NUMPOPS; i++))\nfor ((i = 0; i < 1; i++))\ndo\n    for ((j = i + 1; j < NUMPOPS; j++))\n    do\n        POP1=${POPSARR[i]}\n        POP2=${POPSARR[j]}\n\n        popcorn fit -v 1 \\\n            --sfile1 $SUMSTATS_PREF/${POP1}_Overall_${SUBSET}_${MAF}.sumstats.txt \\\n            --sfile2 $SUMSTATS_PREF/${POP2}_Overall_${SUBSET}_${MAF}.sumstats.txt \\\n            --cfile $SCORES_PREF/${POP1}_${POP2}_cleaned_scores.txt \\\n            ${POP1}_${POP2}_Overall_${SUBSET}_${MAF}.txt\n        \n        dx upload ${POP1}_${POP2}_Overall_${SUBSET}_${MAF}.txt*\n    done\ndone", "metadata": {"trusted": true, "tags": []}, "execution_count": 10, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_Overall_HM3_05.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AMR_Overall_HM3_05.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_AMR_cleaned_scores.txt AFR_AMR_Overall_HM3_05.txt\nBeginning analysis at Mon Jul 22 15:46:04 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n978211 934878 SNPs detected in input.\n782508 SNPs remaining after merging with score file.\n782508 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.399124 NaN\nh2^2   0.680149 NaN\npg     0.317072 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3912 .\nJackknife iter: 200 \n      Val (obs)        SE         Z     P (Z)\nh1^2   0.399124  0.090005  4.434466  0.000009\nh2^2   0.680149  0.365613  1.860295  0.062844\npgi    0.317072  0.241995  2.822072  0.004771\nAnalysis finished at Mon Jul 22 15:49:48 2024\nTotal time elapsed: 223.98\n[===========================================================>] Uploaded 270 of 270 bytes (100%) AFR_AMR_Overall_HM3_05.txt\nID                          file-GpP7z784BzFz7XXPbp4FF42p\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_Overall_HM3_05.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jul 22 15:49:49 2024\nCreated by                  omjahagirdar\n via the job                job-GpP5PK84BzFb0P0FX236Q30y\nLast modified               Mon Jul 22 15:49:50 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GpP7z7Q4BzFz7XXPbp4FF42y\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_Overall_HM3_05.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jul 22 15:49:50 2024\nCreated by                  omjahagirdar\n via the job                job-GpP5PK84BzFb0P0FX236Q30y\nLast modified               Mon Jul 22 15:49:50 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,576 of 5,576 bytes (100%) AFR_AMR_Overall_HM3_05.txt.o.log\nID                          file-GpP7z7Q4BzFxGYGqVF01xb0f\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_Overall_HM3_05.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jul 22 15:49:50 2024\nCreated by                  omjahagirdar\n via the job                job-GpP5PK84BzFb0P0FX236Q30y\nLast modified               Mon Jul 22 15:49:51 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_Overall_HM3_05.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EAS_Overall_HM3_05.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_EAS_cleaned_scores.txt AFR_EAS_Overall_HM3_05.txt\nBeginning analysis at Mon Jul 22 15:49:52 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n978211 861724 SNPs detected in input.\n716686 SNPs remaining after merging with score file.\n716686 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.336194 NaN\nh2^2   0.385032 NaN\npg     0.498674 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3583 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.336194  0.080134  4.195390  2.724029e-05\nh2^2   0.385032  0.070422  5.467503  4.564186e-08\npgi    0.498674  0.174382  2.874869  4.041954e-03\nAnalysis finished at Mon Jul 22 15:53:20 2024\nTotal time elapsed: 208.12\n[===========================================================>] Uploaded 275 of 275 bytes (100%) AFR_EAS_Overall_HM3_05.txt\nID                          file-GpP80v84BzFfV69ybv7KVXP6\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_Overall_HM3_05.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jul 22 15:53:21 2024\nCreated by                  omjahagirdar\n via the job                job-GpP5PK84BzFb0P0FX236Q30y\nLast modified               Mon Jul 22 15:53:22 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GpP80vQ4BzFj54bZ6QV6q606\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_Overall_HM3_05.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jul 22 15:53:22 2024\nCreated by                  omjahagirdar\n via the job                job-GpP5PK84BzFb0P0FX236Q30y\nLast modified               Mon Jul 22 15:53:23 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,592 of 5,592 bytes (100%) AFR_EAS_Overall_HM3_05.txt.o.log\nID                          file-GpP80vj4BzFgP0B7Y127X4j6\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_Overall_HM3_05.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jul 22 15:53:23 2024\nCreated by                  omjahagirdar\n via the job                job-GpP5PK84BzFb0P0FX236Q30y\nLast modified               Mon Jul 22 15:53:24 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_Overall_HM3_05.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_Overall_HM3_05.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_EUR_cleaned_scores.txt AFR_EUR_Overall_HM3_05.txt\nBeginning analysis at Mon Jul 22 15:53:25 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n978211 968495 SNPs detected in input.\n798056 SNPs remaining after merging with score file.\n798056 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.331302 NaN\nh2^2   0.335972 NaN\npg     0.469547 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3990 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.331302  0.084107  3.939079  8.179510e-05\nh2^2   0.335972  0.064265  5.227896  1.714499e-07\npgi    0.469547  0.135481  3.915330  9.028077e-05\nAnalysis finished at Mon Jul 22 15:57:20 2024\nTotal time elapsed: 234.77\n[===========================================================>] Uploaded 275 of 275 bytes (100%) AFR_EUR_Overall_HM3_05.txt\nID                          file-GpP82j04BzFj54bZ6QV6q6gK\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_Overall_HM3_05.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jul 22 15:57:20 2024\nCreated by                  omjahagirdar\n via the job                job-GpP5PK84BzFb0P0FX236Q30y\nLast modified               Mon Jul 22 15:57:21 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GpP82j84BzFfV69ybv7KVXyF\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_Overall_HM3_05.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jul 22 15:57:21 2024\nCreated by                  omjahagirdar\n via the job                job-GpP5PK84BzFb0P0FX236Q30y\nLast modified               Mon Jul 22 15:57:22 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,592 of 5,592 bytes (100%) AFR_EUR_Overall_HM3_05.txt.o.log\nID                          file-GpP82jQ4BzFgP0B7Y127X554\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_Overall_HM3_05.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jul 22 15:57:22 2024\nCreated by                  omjahagirdar\n via the job                job-GpP5PK84BzFb0P0FX236Q30y\nLast modified               Mon Jul 22 15:57:23 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "POPS=\"AFR AMR EAS EUR\"\n\nread -r -a POPSARR <<< \"$POPS\"\nNUMPOPS=${#POPSARR[@]}\n\nfor ((i = 0; i < NUMPOPS; i++))\ndo\n    for ((j = i + 1; j < NUMPOPS; j++))\n    do\n        POP1=${POPSARR[i]}\n        POP2=${POPSARR[j]}\n\n        popcorn fit -v 1 \\\n            --sfile1 $SUMSTATS_PREF/${POP1}_meta_update.sumstats.txt \\\n            --sfile2 $SUMSTATS_PREF/${POP2}_meta_update.sumstats.txt \\\n            --cfile $SCORES_PREF/${POP1}_${POP2}_cleaned_scores.txt \\\n            --gen_effect \\\n            ${POP1}_${POP2}_meta_corr_update_gen.txt\n        \n        dx upload ${POP1}_${POP2}_meta_corr_update_gen.txt*\n    done\ndone", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AMR_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_AMR_cleaned_scores.txt --gen_effect AFR_AMR_meta_corr_update_gen.txt\nBeginning analysis at Mon Jun 10 20:47:07 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n968675 934878 SNPs detected in input.\n732036 SNPs remaining after merging with score file.\n732035 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.287118 NaN\nh2^2   0.647640 NaN\npg     0.259431 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3660 .\nJackknife iter: 200 \n      Val (obs)        SE         Z     P (Z)\nh1^2   0.287118  0.156512  1.834473  0.066584\nh2^2   0.647640  0.375281  1.725747  0.084393\npge    0.259431  0.500871  1.478562  0.139257\nAnalysis finished at Mon Jun 10 20:50:32 2024\nTotal time elapsed: 204.72\n[===========================================================>] Uploaded 265 of 265 bytes (100%) AFR_AMR_meta_corr_update_gen.txt\nID                          file-GkYfK604BzFbg8QfyzVjXZV5\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_meta_corr_update_gen.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 20:50:32 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 20:50:33 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GkYfK6Q4BzFpV639q0k8qjvV\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_meta_corr_update_gen.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 20:50:34 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 20:50:34 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,589 of 5,589 bytes (100%) AFR_AMR_meta_corr_update_gen.txt.o.log\nID                          file-GkYfK6Q4BzFq2z74gp6fxPk5\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_meta_corr_update_gen.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 20:50:34 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 20:50:35 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EAS_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_EAS_cleaned_scores.txt --gen_effect AFR_EAS_meta_corr_update_gen.txt\nBeginning analysis at Mon Jun 10 20:50:36 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n968675 861724 SNPs detected in input.\n681336 SNPs remaining after merging with score file.\n681336 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.232736 NaN\nh2^2   0.398550 NaN\npg     0.712405 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3406 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.232736  0.135474  1.717942  8.580715e-02\nh2^2   0.398550  0.074525  5.347851  8.900488e-08\npge    0.712405  0.347144  0.828461  4.074095e-01\nAnalysis finished at Mon Jun 10 20:53:51 2024\nTotal time elapsed: 194.68\n[===========================================================>] Uploaded 267 of 267 bytes (100%) AFR_EAS_meta_corr_update_gen.txt\nID                          file-GkYfPj04BzFbg8QfyzVjXZk5\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_meta_corr_update_gen.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 20:53:52 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 20:53:53 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GkYfPj84BzFX57JpJjY6vQ1V\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_meta_corr_update_gen.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 20:53:53 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 20:53:53 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,605 of 5,605 bytes (100%) AFR_EAS_meta_corr_update_gen.txt.o.log\nID                          file-GkYfPjQ4BzFbg8QfyzVjXZk7\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_meta_corr_update_gen.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 20:53:54 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 20:53:54 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_EUR_cleaned_scores.txt --gen_effect AFR_EUR_meta_corr_update_gen.txt\nBeginning analysis at Mon Jun 10 20:53:55 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n968675 968495 SNPs detected in input.\n756079 SNPs remaining after merging with score file.\n756077 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.186400 NaN\nh2^2   0.316670 NaN\npg     0.731275 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3780 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.186400  0.148098  1.258622  2.081671e-01\nh2^2   0.316670  0.059474  5.324478  1.012435e-07\npge    0.731275  0.333804  0.805037  4.207981e-01\nAnalysis finished at Mon Jun 10 20:57:37 2024\nTotal time elapsed: 221.19\n[===========================================================>] Uploaded 270 of 270 bytes (100%) AFR_EUR_meta_corr_update_gen.txt\nID                          file-GkYfVQ84BzFq4xPYXJjXYK4y\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_corr_update_gen.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 20:57:37 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 20:57:38 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GkYfVQQ4BzFX57JpJjY6vQPG\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_corr_update_gen.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 20:57:38 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 20:57:39 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,605 of 5,605 bytes (100%) AFR_EUR_meta_corr_update_gen.txt.o.log\nID                          file-GkYfVQj4BzFq4xPYXJjXYK55\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_corr_update_gen.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 20:57:39 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 20:57:40 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AMR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EAS_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AMR_EAS_cleaned_scores.txt --gen_effect AMR_EAS_meta_corr_update_gen.txt\nBeginning analysis at Mon Jun 10 20:57:41 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n934878 861724 SNPs detected in input.\n732256 SNPs remaining after merging with score file.\n732255 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.701500 NaN\nh2^2   0.404317 NaN\npg     0.418712 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3661 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.701500  0.348085  2.015315  4.387161e-02\nh2^2   0.404317  0.071251  5.674572  1.390357e-08\npge    0.418712  0.235202  2.471443  1.345689e-02\nAnalysis finished at Mon Jun 10 21:01:06 2024\nTotal time elapsed: 205.06\n[===========================================================>] Uploaded 267 of 267 bytes (100%) AMR_EAS_meta_corr_update_gen.txt\nID                          file-GkYfY4j4BzFY415XYf01p9g8\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EAS_meta_corr_update_gen.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 21:01:07 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 21:01:08 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GkYfY504BzFx5B5f88v3JqZ9\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EAS_meta_corr_update_gen.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 21:01:08 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 21:01:09 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,605 of 5,605 bytes (100%) AMR_EAS_meta_corr_update_gen.txt.o.log\nID                          file-GkYfY584BzFYZ5BQFB7zGgpZ\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EAS_meta_corr_update_gen.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 21:01:09 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 21:01:10 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AMR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AMR_EUR_cleaned_scores.txt --gen_effect AMR_EUR_meta_corr_update_gen.txt\nBeginning analysis at Mon Jun 10 21:01:11 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n934878 968495 SNPs detected in input.\n797615 SNPs remaining after merging with score file.\n797615 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.671796 NaN\nh2^2   0.366174 NaN\npg     0.675411 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3988 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.671796  0.355531  1.889557  5.881724e-02\nh2^2   0.366174  0.068743  5.326680  1.000243e-07\npge    0.675411  0.219322  1.479965  1.388826e-01\nAnalysis finished at Mon Jun 10 21:05:02 2024\nTotal time elapsed: 230.89\n[===========================================================>] Uploaded 270 of 270 bytes (100%) AMR_EUR_meta_corr_update_gen.txt\nID                          file-GkYfZzQ4BzFXGbZbjqFxV4bZ\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EUR_meta_corr_update_gen.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 21:05:02 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 21:05:03 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GkYfb004BzFXGbZbjqFxV4by\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EUR_meta_corr_update_gen.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 21:05:04 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 21:05:05 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,605 of 5,605 bytes (100%) AMR_EUR_meta_corr_update_gen.txt.o.log\nID                          file-GkYfb0Q4BzFkB4QBGqjgg3Xk\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EUR_meta_corr_update_gen.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 21:05:06 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 21:05:07 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EAS_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/EAS_EUR_cleaned_scores.txt --gen_effect EAS_EUR_meta_corr_update_gen.txt\nBeginning analysis at Mon Jun 10 21:05:08 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n861724 968495 SNPs detected in input.\n746651 SNPs remaining after merging with score file.\n746651 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.353555 NaN\nh2^2   0.324556 NaN\npg     0.806421 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3733 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.353555  0.068523  5.159689  2.473607e-07\nh2^2   0.324556  0.059257  5.477092  4.323718e-08\npge    0.806421  0.090506  2.138851  3.244772e-02\nAnalysis finished at Mon Jun 10 21:08:32 2024\nTotal time elapsed: 204.38\n[===========================================================>] Uploaded 271 of 271 bytes (100%) EAS_EUR_meta_corr_update_gen.txt\nID                          file-GkYffZ84BzFVzyv59qBkk43q\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        EAS_EUR_meta_corr_update_gen.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 21:08:33 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 21:08:33 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GkYffZQ4BzFkB4QBGqjgg461\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        EAS_EUR_meta_corr_update_gen.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 21:08:34 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 21:08:34 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,605 of 5,605 bytes (100%) EAS_EUR_meta_corr_update_gen.txt.o.log\nID                          file-GkYffZj4BzFVzyv59qBkk43z\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        EAS_EUR_meta_corr_update_gen.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Jun 10 21:08:35 2024\nCreated by                  omjahagirdar\n via the job                job-GkYf7K04BzFv5BbX36yG2fjj\nLast modified               Mon Jun 10 21:08:35 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "## Overall Breast Cancer", "metadata": {"tags": []}}, {"cell_type": "code", "source": "POPS=\"AFR AMR EAS EUR\"\n\nread -r -a POPSARR <<< \"$POPS\"\nNUMPOPS=${#POPSARR[@]}\n\nfor ((i = 0; i < NUMPOPS; i++))\ndo\n    for ((j = i + 1; j < NUMPOPS; j++))\n    do\n        POP1=${POPSARR[i]}\n        POP2=${POPSARR[j]}\n\n        popcorn fit -v 1 \\\n            --sfile1 $SUMSTATS_PREF/${POP1}_meta_update.sumstats.txt \\\n            --sfile2 $SUMSTATS_PREF/${POP2}_meta_update.sumstats.txt \\\n            --cfile $SCORES_PREF/${POP1}_${POP2}_cleaned_scores.txt \\\n            ${POP1}_${POP2}_meta_corr_update.txt\n        \n        dx upload ${POP1}_${POP2}_meta_corr_update.txt*\n    done\ndone", "metadata": {"trusted": true, "tags": []}, "execution_count": 9, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AMR_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_AMR_cleaned_scores.txt AFR_AMR_meta_corr_update.txt\nBeginning analysis at Thu May  2 00:21:09 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\n968675 934878 SNPs detected in input.\n732036 SNPs remaining after merging with score file.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:191: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n  if not data1['a1'][0].isupper():\n732035 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.287118 NaN\nh2^2   0.647640 NaN\npg     0.259431 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3660 .\nJackknife iter: 200 \n/opt/notebooks/Popcorn/popcorn/fit.py:249: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\nYou are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\nA typical example is when you are setting values in a column of a DataFrame, like:\n\ndf[\"col\"][row_indexer] = value\n\nUse `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n\n  self.res['Z']['pg'] = (1.0-self.res['Val (obs)']['pg'])/self.res['SE']['pg']\n      Val (obs)        SE         Z     P (Z)\nh1^2   0.287118  0.156512  1.834473  0.066584\nh2^2   0.647640  0.375281  1.725747  0.084393\npgi    0.259431  0.500871  1.478562  0.139257\nAnalysis finished at Thu May  2 00:25:00 2024\nTotal time elapsed: 230.76\n[===========================================================>] Uploaded 265 of 265 bytes (100%) AFR_AMR_meta_corr_update.txt\nID                          file-GjkJpg84BzFg3YYfQz67PvXV\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_meta_corr_update.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:25:01 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:25:02 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,021 of 2,021 bytes (100%) AFR_AMR_meta_corr_update.txt.e.log\nID                          file-GjkJpgQ4BzFj2jv3f9J76P6G\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_meta_corr_update.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:25:02 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:25:03 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 8,193 of 8,193 bytes (100%) AFR_AMR_meta_corr_update.txt.o.log\nID                          file-GjkJpgj4BzFqbVj6xV1Gb7J5\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_AMR_meta_corr_update.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:25:03 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:25:04 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EAS_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_EAS_cleaned_scores.txt AFR_EAS_meta_corr_update.txt\nBeginning analysis at Thu May  2 00:25:05 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\n968675 861724 SNPs detected in input.\n681336 SNPs remaining after merging with score file.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:191: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n  if not data1['a1'][0].isupper():\n681336 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.232736 NaN\nh2^2   0.398550 NaN\npg     0.712405 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3406 .\nJackknife iter: 200 \n/opt/notebooks/Popcorn/popcorn/fit.py:249: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\nYou are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\nA typical example is when you are setting values in a column of a DataFrame, like:\n\ndf[\"col\"][row_indexer] = value\n\nUse `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n\n  self.res['Z']['pg'] = (1.0-self.res['Val (obs)']['pg'])/self.res['SE']['pg']\n      Val (obs)        SE         Z         P (Z)\nh1^2   0.232736  0.135474  1.717942  8.580715e-02\nh2^2   0.398550  0.074525  5.347851  8.900488e-08\npgi    0.712405  0.347144  0.828461  4.074095e-01\nAnalysis finished at Thu May  2 00:28:34 2024\nTotal time elapsed: 209.36\n[===========================================================>] Uploaded 267 of 267 bytes (100%) AFR_EAS_meta_corr_update.txt\nID                          file-GjkJvGj4BzFVj65qQxqYy9q6\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_meta_corr_update.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:28:35 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:28:36 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,021 of 2,021 bytes (100%) AFR_EAS_meta_corr_update.txt.e.log\nID                          file-GjkJvJ04BzFj2jv3f9J76PGq\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_meta_corr_update.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:28:36 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:28:37 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 8,209 of 8,209 bytes (100%) AFR_EAS_meta_corr_update.txt.o.log\nID                          file-GjkJvJ84BzFY6bBX56gxGqKG\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_meta_corr_update.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:28:37 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:28:38 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_EUR_cleaned_scores.txt AFR_EUR_meta_corr_update.txt\nBeginning analysis at Thu May  2 00:28:39 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\n968675 968495 SNPs detected in input.\n756079 SNPs remaining after merging with score file.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:191: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n  if not data1['a1'][0].isupper():\n756077 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.186400 NaN\nh2^2   0.316670 NaN\npg     0.731275 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3780 .\nJackknife iter: 200 \n/opt/notebooks/Popcorn/popcorn/fit.py:249: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\nYou are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\nA typical example is when you are setting values in a column of a DataFrame, like:\n\ndf[\"col\"][row_indexer] = value\n\nUse `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n\n  self.res['Z']['pg'] = (1.0-self.res['Val (obs)']['pg'])/self.res['SE']['pg']\n      Val (obs)        SE         Z         P (Z)\nh1^2   0.186400  0.148098  1.258622  2.081671e-01\nh2^2   0.316670  0.059474  5.324478  1.012435e-07\npgi    0.731275  0.333804  0.805037  4.207981e-01\nAnalysis finished at Thu May  2 00:32:34 2024\nTotal time elapsed: 235.12\n[===========================================================>] Uploaded 270 of 270 bytes (100%) AFR_EUR_meta_corr_update.txt\nID                          file-GjkJy8j4BzFz5vX57b7q7k02\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_corr_update.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:32:35 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:32:36 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,021 of 2,021 bytes (100%) AFR_EUR_meta_corr_update.txt.e.log\nID                          file-GjkJy904BzFqbVj6xV1Gb867\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_corr_update.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:32:36 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:32:37 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 8,209 of 8,209 bytes (100%) AFR_EUR_meta_corr_update.txt.o.log\nID                          file-GjkJy984BzFg3YYfQz67Px84\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_corr_update.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:32:37 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:32:38 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AMR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EAS_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AMR_EAS_cleaned_scores.txt AMR_EAS_meta_corr_update.txt\nBeginning analysis at Thu May  2 00:32:39 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\n934878 861724 SNPs detected in input.\n732256 SNPs remaining after merging with score file.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:191: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n  if not data1['a1'][0].isupper():\n732255 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.701500 NaN\nh2^2   0.404317 NaN\npg     0.418712 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3661 .\nJackknife iter: 200 \n/opt/notebooks/Popcorn/popcorn/fit.py:249: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\nYou are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\nA typical example is when you are setting values in a column of a DataFrame, like:\n\ndf[\"col\"][row_indexer] = value\n\nUse `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n\n  self.res['Z']['pg'] = (1.0-self.res['Val (obs)']['pg'])/self.res['SE']['pg']\n      Val (obs)        SE         Z         P (Z)\nh1^2   0.701500  0.348085  2.015315  4.387161e-02\nh2^2   0.404317  0.071251  5.674572  1.390357e-08\npgi    0.418712  0.235202  2.471443  1.345689e-02\nAnalysis finished at Thu May  2 00:36:24 2024\nTotal time elapsed: 224.66\n[===========================================================>] Uploaded 267 of 267 bytes (100%) AMR_EAS_meta_corr_update.txt\nID                          file-GjkK0284BzFv3Qg72310vy9F\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EAS_meta_corr_update.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:36:25 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:36:26 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,021 of 2,021 bytes (100%) AMR_EAS_meta_corr_update.txt.e.log\nID                          file-GjkK02Q4BzFz3xqJV8V56B8Y\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EAS_meta_corr_update.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:36:26 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:36:27 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 8,209 of 8,209 bytes (100%) AMR_EAS_meta_corr_update.txt.o.log\nID                          file-GjkK02j4BzFq6YfjGQ7ZXxp0\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EAS_meta_corr_update.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:36:27 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:36:28 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AMR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AMR_EUR_cleaned_scores.txt AMR_EUR_meta_corr_update.txt\nBeginning analysis at Thu May  2 00:36:29 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\n934878 968495 SNPs detected in input.\n797615 SNPs remaining after merging with score file.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:191: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n  if not data1['a1'][0].isupper():\n797615 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.671796 NaN\nh2^2   0.366174 NaN\npg     0.675411 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3988 .\nJackknife iter: 200 \n/opt/notebooks/Popcorn/popcorn/fit.py:249: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\nYou are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\nA typical example is when you are setting values in a column of a DataFrame, like:\n\ndf[\"col\"][row_indexer] = value\n\nUse `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n\n  self.res['Z']['pg'] = (1.0-self.res['Val (obs)']['pg'])/self.res['SE']['pg']\n      Val (obs)        SE         Z         P (Z)\nh1^2   0.671796  0.355531  1.889557  5.881724e-02\nh2^2   0.366174  0.068743  5.326680  1.000243e-07\npgi    0.675411  0.219322  1.479965  1.388826e-01\nAnalysis finished at Thu May  2 00:40:39 2024\nTotal time elapsed: 249.7\n[===========================================================>] Uploaded 270 of 270 bytes (100%) AMR_EUR_meta_corr_update.txt\nID                          file-GjkK21j4BzFZPG8Yp0yf214p\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EUR_meta_corr_update.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:40:39 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:40:40 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,021 of 2,021 bytes (100%) AMR_EUR_meta_corr_update.txt.e.log\nID                          file-GjkK2204BzFZPG8Yp0yf214v\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EUR_meta_corr_update.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:40:40 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:40:41 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 8,208 of 8,208 bytes (100%) AMR_EUR_meta_corr_update.txt.o.log\nID                          file-GjkK22Q4BzFj2FP8184X4y84\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AMR_EUR_meta_corr_update.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:40:42 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:40:43 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EAS_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_update.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/EAS_EUR_cleaned_scores.txt EAS_EUR_meta_corr_update.txt\nBeginning analysis at Thu May  2 00:40:44 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\n861724 968495 SNPs detected in input.\n746651 SNPs remaining after merging with score file.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:191: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n  if not data1['a1'][0].isupper():\n746651 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.353555 NaN\nh2^2   0.324556 NaN\npg     0.806421 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 3733 .\nJackknife iter: 200 \n/opt/notebooks/Popcorn/popcorn/fit.py:249: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\nYou are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\nA typical example is when you are setting values in a column of a DataFrame, like:\n\ndf[\"col\"][row_indexer] = value\n\nUse `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n\n  self.res['Z']['pg'] = (1.0-self.res['Val (obs)']['pg'])/self.res['SE']['pg']\n      Val (obs)        SE         Z         P (Z)\nh1^2   0.353555  0.068523  5.159689  2.473607e-07\nh2^2   0.324556  0.059257  5.477092  4.323718e-08\npgi    0.806421  0.090506  2.138851  3.244772e-02\nAnalysis finished at Thu May  2 00:44:32 2024\nTotal time elapsed: 228.26\n[===========================================================>] Uploaded 271 of 271 bytes (100%) EAS_EUR_meta_corr_update.txt\nID                          file-GjkK3v84BzFqJy66FP5GpVX1\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        EAS_EUR_meta_corr_update.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:44:33 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:44:33 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,021 of 2,021 bytes (100%) EAS_EUR_meta_corr_update.txt.e.log\nID                          file-GjkK3vQ4BzFVZ77YZJFvp184\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        EAS_EUR_meta_corr_update.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:44:34 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:44:35 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 8,209 of 8,209 bytes (100%) EAS_EUR_meta_corr_update.txt.o.log\nID                          file-GjkK3vj4BzFkV3v999gxjPff\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        EAS_EUR_meta_corr_update.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Thu May  2 00:44:35 2024\nCreated by                  omjahagirdar\n via the job                job-GjkG0X04BzFq8Y4X29fjYqFf\nLast modified               Thu May  2 00:44:36 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "dx ls", "metadata": {"trusted": true, "tags": []}, "execution_count": 10, "outputs": [{"name": "stdout", "text": "AFR_AMR_meta_corr_update.txt\nAFR_AMR_meta_corr_update.txt.e.log\nAFR_AMR_meta_corr_update.txt.o.log\nAFR_EAS_meta_corr.txt\nAFR_EAS_meta_corr.txt.e.log\nAFR_EAS_meta_corr.txt.o.log\nAFR_EAS_meta_corr_update.txt\nAFR_EAS_meta_corr_update.txt.e.log\nAFR_EAS_meta_corr_update.txt.o.log\nAFR_EAS_meta_ERneg_corr_update.txt\nAFR_EAS_meta_ERneg_corr_update.txt.e.log\nAFR_EAS_meta_ERneg_corr_update.txt.o.log\nAFR_EAS_meta_ERpos_corr_update.txt\nAFR_EAS_meta_ERpos_corr_update.txt.e.log\nAFR_EAS_meta_ERpos_corr_update.txt.o.log\nAFR_EUR_corr_preliminary.txt\nAFR_EUR_meta_corr.txt\nAFR_EUR_meta_corr.txt.e.log\nAFR_EUR_meta_corr.txt.o.log\nAFR_EUR_meta_corr_update.txt\nAFR_EUR_meta_corr_update.txt.e.log\nAFR_EUR_meta_corr_update.txt.o.log\nAFR_EUR_meta_ERneg_corr.txt\nAFR_EUR_meta_ERneg_corr.txt.e.log\nAFR_EUR_meta_ERneg_corr.txt.o.log\nAFR_EUR_meta_ERneg_corr_update.txt\nAFR_EUR_meta_ERneg_corr_update.txt.e.log\nAFR_EUR_meta_ERneg_corr_update.txt.o.log\nAFR_EUR_meta_ERneg_totsamples_corr.txt\nAFR_EUR_meta_ERneg_totsamples_corr.txt.e.log\nAFR_EUR_meta_ERneg_totsamples_corr.txt.o.log\nAFR_EUR_meta_ERpos_corr.txt\nAFR_EUR_meta_ERpos_corr.txt.e.log\nAFR_EUR_meta_ERpos_corr.txt.o.log\nAFR_EUR_meta_ERpos_corr_update.txt\nAFR_EUR_meta_ERpos_corr_update.txt.e.log\nAFR_EUR_meta_ERpos_corr_update.txt.o.log\nAFR_EUR_meta_ERpos_totsamples_corr.txt\nAFR_EUR_meta_ERpos_totsamples_corr.txt.e.log\nAFR_EUR_meta_ERpos_totsamples_corr.txt.o.log\nAMR_EAS_meta_corr_update.txt\nAMR_EAS_meta_corr_update.txt.e.log\nAMR_EAS_meta_corr_update.txt.o.log\nAMR_EUR_meta_corr_update.txt\nAMR_EUR_meta_corr_update.txt.e.log\nAMR_EUR_meta_corr_update.txt.o.log\nEAS_EUR_meta_corr.txt\nEAS_EUR_meta_corr.txt.e.log\nEAS_EUR_meta_corr.txt.o.log\nEAS_EUR_meta_corr_update.txt\nEAS_EUR_meta_corr_update.txt.e.log\nEAS_EUR_meta_corr_update.txt.o.log\nEAS_EUR_meta_ERneg_corr_update.txt\nEAS_EUR_meta_ERneg_corr_update.txt.e.log\nEAS_EUR_meta_ERneg_corr_update.txt.o.log\nEAS_EUR_meta_ERpos_corr_update.txt\nEAS_EUR_meta_ERpos_corr_update.txt.e.log\nEAS_EUR_meta_ERpos_corr_update.txt.o.log\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "POPS=\"AMR EAS EUR\"\n\nread -r -a POPSARR <<< \"$POPS\"\nNUMPOPS=${#POPSARR[@]}\n\nfor ((i = 0; i < NUMPOPS; i++))\ndo\n    POP=${POPSARR[i]}\n\n    popcorn fit -v 1 \\\n        --sfile1 $SUMSTATS_PREF/AFR_meta_update.sumstats.txt \\\n        --sfile2 $SUMSTATS_PREF/${POP}_meta_update.sumstats.txt \\\n        --cfile $SCORES_PREF/AA_${POP}_cleaned_scores.txt \\\n        AA_${POP}_meta_corr_update.txt\n\n    dx upload AA_${POP}_meta_corr_update.txt*\ndone", "metadata": {"trusted": true, "tags": []}, "execution_count": 8, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AMR_meta_update.sumstats.txt --cfile /mnt/project/Popcorn/AFR_AMR AA_AMR_meta_corr_update.txt\nBeginning analysis at Sat May  4 18:02:01 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\n968675 934878 SNPs detected in input.\n0 SNPs remaining after merging with score file.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:191: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n  if not data1['a1'][0].isupper():\nTraceback (most recent call last):\n  File \"/opt/conda/bin/popcorn\", line 8, in <module>\n    sys.exit(main())\n  File \"/opt/notebooks/Popcorn/popcorn/__main__.py\", line 206, in main\n    data = sumstats.sumstats_2_trait(scores,args)\n  File \"/opt/notebooks/Popcorn/popcorn/sumstats.py\", line 191, in __init__\n    if not data1['a1'][0].isupper():\n  File \"/opt/conda/lib/python3.9/site-packages/pandas/core/series.py\", line 1118, in __getitem__\n    return self._values[key]\nIndexError: index 0 is out of bounds for axis 0 with size 0\n[===========================================================>] Uploaded 1,239 of 1,239 bytes (100%) AA_AMR_meta_corr_update.txt.e.log\nID                          file-Gjq7QF84BzFy0Xp8Zk8qxYgp\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AA_AMR_meta_corr_update.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Sat May  4 18:02:21 2024\nCreated by                  omjahagirdar\n via the job                job-Gjq78K04BzFp1GfKYFvX6pqX\nLast modified               Sat May  4 18:02:21 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,126 of 2,126 bytes (100%) AA_AMR_meta_corr_update.txt.o.log\nID                          file-Gjq7QFQ4BzFp71QqyGJ9zZb8\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AA_AMR_meta_corr_update.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Sat May  4 18:02:22 2024\nCreated by                  omjahagirdar\n via the job                job-Gjq78K04BzFp1GfKYFvX6pqX\nLast modified               Sat May  4 18:02:22 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EAS_meta_update.sumstats.txt --cfile /mnt/project/Popcorn/AFR_EAS AA_EAS_meta_corr_update.txt\nBeginning analysis at Sat May  4 18:02:24 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\n968675 861724 SNPs detected in input.\n0 SNPs remaining after merging with score file.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:191: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n  if not data1['a1'][0].isupper():\nTraceback (most recent call last):\n  File \"/opt/conda/bin/popcorn\", line 8, in <module>\n    sys.exit(main())\n  File \"/opt/notebooks/Popcorn/popcorn/__main__.py\", line 206, in main\n    data = sumstats.sumstats_2_trait(scores,args)\n  File \"/opt/notebooks/Popcorn/popcorn/sumstats.py\", line 191, in __init__\n    if not data1['a1'][0].isupper():\n  File \"/opt/conda/lib/python3.9/site-packages/pandas/core/series.py\", line 1118, in __getitem__\n    return self._values[key]\nIndexError: index 0 is out of bounds for axis 0 with size 0\n[===========================================================>] Uploaded 1,239 of 1,239 bytes (100%) AA_EAS_meta_corr_update.txt.e.log\nID                          file-Gjq7QQj4BzFpfXppJQvFZFz8\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AA_EAS_meta_corr_update.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Sat May  4 18:02:43 2024\nCreated by                  omjahagirdar\n via the job                job-Gjq78K04BzFp1GfKYFvX6pqX\nLast modified               Sat May  4 18:02:44 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,126 of 2,126 bytes (100%) AA_EAS_meta_corr_update.txt.o.log\nID                          file-Gjq7QV04BzFXP2zZYzpZx1zG\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AA_EAS_meta_corr_update.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Sat May  4 18:02:44 2024\nCreated by                  omjahagirdar\n via the job                job-Gjq78K04BzFp1GfKYFvX6pqX\nLast modified               Sat May  4 18:02:45 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\nPopcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_update.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_update.sumstats.txt --cfile /mnt/project/Popcorn/AFR_EUR AA_EUR_meta_corr_update.txt\nBeginning analysis at Sat May  4 18:02:46 2024\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:104: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n  has_comp = data[['a1', 'a2']].applymap(lambda x: x in compliment)\n968675 968495 SNPs detected in input.\n0 SNPs remaining after merging with score file.\n/opt/notebooks/Popcorn/popcorn/sumstats.py:191: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n  if not data1['a1'][0].isupper():\nTraceback (most recent call last):\n  File \"/opt/conda/bin/popcorn\", line 8, in <module>\n    sys.exit(main())\n  File \"/opt/notebooks/Popcorn/popcorn/__main__.py\", line 206, in main\n    data = sumstats.sumstats_2_trait(scores,args)\n  File \"/opt/notebooks/Popcorn/popcorn/sumstats.py\", line 191, in __init__\n    if not data1['a1'][0].isupper():\n  File \"/opt/conda/lib/python3.9/site-packages/pandas/core/series.py\", line 1118, in __getitem__\n    return self._values[key]\nIndexError: index 0 is out of bounds for axis 0 with size 0\n[===========================================================>] Uploaded 1,239 of 1,239 bytes (100%) AA_EUR_meta_corr_update.txt.e.log\nID                          file-Gjq7Qf84BzFvQGP9j2jxkbZv\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AA_EUR_meta_corr_update.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Sat May  4 18:03:05 2024\nCreated by                  omjahagirdar\n via the job                job-Gjq78K04BzFp1GfKYFvX6pqX\nLast modified               Sat May  4 18:03:05 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,126 of 2,126 bytes (100%) AA_EUR_meta_corr_update.txt.o.log\nID                          file-Gjq7QfQ4BzFvQGP9j2jxkbZz\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AA_EUR_meta_corr_update.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Sat May  4 18:03:06 2024\nCreated by                  omjahagirdar\n via the job                job-Gjq78K04BzFp1GfKYFvX6pqX\nLast modified               Sat May  4 18:03:07 2024\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}]}