{"metadata": {"language_info": {"name": "bash", "codemirror_mode": "shell", "mimetype": "text/x-sh", "file_extension": ".sh"}, "kernelspec": {"name": "bash", "display_name": "Bash", "language": "bash"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "markdown", "source": "This bash script runs Popcorn, a program for estimating the correlation of causal variant effect, on the Pre-Confluence Project Data.", "metadata": {"tags": []}}, {"cell_type": "code", "source": "dx cd\ndx cd Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/corr_files\ndx ls", "metadata": {"trusted": true, "tags": []}, "execution_count": 1, "outputs": [{"name": "stdout", "text": "AFR_EAS_meta_corr.txt\nAFR_EAS_meta_corr.txt.e.log\nAFR_EAS_meta_corr.txt.o.log\nAFR_EUR_corr_preliminary.txt\nAFR_EUR_meta_corr.txt\nAFR_EUR_meta_corr.txt.e.log\nAFR_EUR_meta_corr.txt.o.log\nAFR_EUR_meta_ERneg_corr.txt : file-GbXZ3X04BzFf3z2JX352B494\nAFR_EUR_meta_ERneg_corr.txt : file-Gbb9yyj4BzFpKgfQXf9XxgZ8\nAFR_EUR_meta_ERneg_corr.txt.e.log\nAFR_EUR_meta_ERneg_corr.txt.o.log\nAFR_EUR_meta_ERpos_corr.txt\nAFR_EUR_meta_ERpos_corr.txt.e.log\nAFR_EUR_meta_ERpos_corr.txt.o.log\nEAS_EUR_meta_corr.txt\nEAS_EUR_meta_corr.txt.e.log\nEAS_EUR_meta_corr.txt.o.log\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "# Install Popcorn", "metadata": {}}, {"cell_type": "code", "source": "python --version    # check running python3", "metadata": {"trusted": true, "tags": []}, "execution_count": 2, "outputs": [{"name": "stdout", "text": "Python 3.9.16\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "git clone https://github.com/brielin/Popcorn.git\ncd Popcorn\npip install .", "metadata": {"trusted": true, "tags": []}, "execution_count": 3, "outputs": [{"name": "stdout", "text": "Cloning into 'Popcorn'...\nremote: Enumerating objects: 639, done.        \nremote: Counting objects: 100% (83/83), done.        \nremote: Compressing objects: 100% (56/56), done.        \nremote: Total 639 (delta 43), reused 61 (delta 26), pack-reused 556        \nReceiving objects: 100% (639/639), 19.28 MiB | 10.20 MiB/s, done.\nResolving deltas: 100% (392/392), done.\n\u001b[33mWARNING: The directory '/home/dnanexus/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\u001b[0m\u001b[33m\n\u001b[0mProcessing /opt/notebooks/Popcorn\n  Preparing metadata (setup.py) ... \u001b[?25ldone\n\u001b[?25hRequirement already satisfied: numpy in /opt/conda/lib/python3.9/site-packages (from popcorn==1.0.0) (1.23.5)\nRequirement already satisfied: scipy in /opt/conda/lib/python3.9/site-packages (from popcorn==1.0.0) (1.9.3)\nRequirement already satisfied: pandas in /opt/conda/lib/python3.9/site-packages (from popcorn==1.0.0) (1.5.3)\nCollecting pysnptools\n  Downloading pysnptools-0.5.11.tar.gz (221 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m221.6/221.6 kB\u001b[0m \u001b[31m253.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25h  Installing build dependencies ... \u001b[?25ldone\n\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n\u001b[?25h  Installing backend dependencies ... \u001b[?25ldone\n\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n\u001b[?25hCollecting bottleneck\n  Downloading Bottleneck-1.3.7-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (353 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m353.1/353.1 kB\u001b[0m \u001b[31m336.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting statsmodels\n  Downloading statsmodels-0.14.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (10.1 MB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m10.1/10.1 MB\u001b[0m \u001b[31m95.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m \u001b[36m0:00:01\u001b[0m\n\u001b[?25hRequirement already satisfied: pytz>=2020.1 in /opt/conda/lib/python3.9/site-packages (from pandas->popcorn==1.0.0) (2023.3.post1)\nRequirement already satisfied: python-dateutil>=2.8.1 in /opt/conda/lib/python3.9/site-packages (from pandas->popcorn==1.0.0) (2.8.2)\nRequirement already satisfied: psutil>=5.6.3 in /opt/conda/lib/python3.9/site-packages (from pysnptools->popcorn==1.0.0) (5.9.6)\nCollecting cloudpickle>=2.2.0\n  Downloading cloudpickle-3.0.0-py3-none-any.whl (20 kB)\nCollecting h5py>=2.10.0\n  Downloading h5py-3.10.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.8 MB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m4.8/4.8 MB\u001b[0m \u001b[31m207.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting cbgen>=1.0.4\n  Downloading cbgen-1.0.4-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (395 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m395.3/395.3 kB\u001b[0m \u001b[31m338.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting bgen-reader>=4.0.8\n  Downloading bgen_reader-4.0.8-py3-none-any.whl (44 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m44.5/44.5 kB\u001b[0m \u001b[31m245.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting bed-reader>=1.0.0\n  Downloading bed_reader-1.0.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.1 MB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m4.1/4.1 MB\u001b[0m \u001b[31m18.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0mm\n\u001b[?25hCollecting more-itertools>=8.7.0\n  Downloading more_itertools-10.1.0-py3-none-any.whl (55 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m55.8/55.8 kB\u001b[0m \u001b[31m10.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hRequirement already satisfied: packaging>=21.3 in /opt/conda/lib/python3.9/site-packages (from statsmodels->popcorn==1.0.0) (23.0)\nCollecting patsy>=0.5.2\n  Downloading patsy-0.5.4-py2.py3-none-any.whl (233 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m233.9/233.9 kB\u001b[0m \u001b[31m333.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hRequirement already satisfied: requests>=1.25.8 in /opt/conda/lib/python3.9/site-packages (from bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (2.28.2)\nRequirement already satisfied: tqdm>=4.43.0 in /opt/conda/lib/python3.9/site-packages (from bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (4.65.0)\nCollecting pytest>=5.4.1\n  Downloading pytest-7.4.3-py3-none-any.whl (325 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m325.1/325.1 kB\u001b[0m \u001b[31m340.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting dask[array,bag,dataframe,delayed]>=2.12.0\n  Downloading dask-2023.12.0-py3-none-any.whl (1.2 MB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m1.2/1.2 MB\u001b[0m \u001b[31m255.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting texttable>=1.6.2\n  Downloading texttable-1.7.0-py2.py3-none-any.whl (10 kB)\nCollecting cachetools>=3.1.1\n  Downloading cachetools-5.3.2-py3-none-any.whl (9.3 kB)\nCollecting xarray>=0.16.0\n  Downloading xarray-2023.11.0-py3-none-any.whl (1.1 MB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m47.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting appdirs>=1.4.3\n  Downloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)\nRequirement already satisfied: cffi in /opt/conda/lib/python3.9/site-packages (from cbgen>=1.0.4->pysnptools->popcorn==1.0.0) (1.15.1)\nRequirement already satisfied: urllib3>=1.26 in /opt/conda/lib/python3.9/site-packages (from cbgen>=1.0.4->pysnptools->popcorn==1.0.0) (1.26.15)\nCollecting pooch\n  Downloading pooch-1.8.0-py3-none-any.whl (62 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m62.7/62.7 kB\u001b[0m \u001b[31m263.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hRequirement already satisfied: six in /opt/conda/lib/python3.9/site-packages (from patsy>=0.5.2->statsmodels->popcorn==1.0.0) (1.16.0)\nRequirement already satisfied: click>=8.1 in /opt/conda/lib/python3.9/site-packages (from dask[array,bag,dataframe,delayed]>=2.12.0->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (8.1.7)\nRequirement already satisfied: pyyaml>=5.3.1 in /opt/conda/lib/python3.9/site-packages (from dask[array,bag,dataframe,delayed]>=2.12.0->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (5.4.1)\nCollecting partd>=1.2.0\n  Downloading partd-1.4.1-py3-none-any.whl (18 kB)\nRequirement already satisfied: toolz>=0.10.0 in /opt/conda/lib/python3.9/site-packages (from dask[array,bag,dataframe,delayed]>=2.12.0->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (0.12.0)\nRequirement already satisfied: importlib-metadata>=4.13.0 in /opt/conda/lib/python3.9/site-packages (from dask[array,bag,dataframe,delayed]>=2.12.0->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (6.8.0)\nCollecting fsspec>=2021.09.0\n  Downloading fsspec-2023.12.1-py3-none-any.whl (168 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m168.9/168.9 kB\u001b[0m \u001b[31m306.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hRequirement already satisfied: exceptiongroup>=1.0.0rc8 in /opt/conda/lib/python3.9/site-packages (from pytest>=5.4.1->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (1.1.3)\nCollecting iniconfig\n  Downloading iniconfig-2.0.0-py3-none-any.whl (5.9 kB)\nRequirement already satisfied: tomli>=1.0.0 in /opt/conda/lib/python3.9/site-packages (from pytest>=5.4.1->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (2.0.1)\nRequirement already satisfied: pluggy<2.0,>=0.12 in /opt/conda/lib/python3.9/site-packages (from pytest>=5.4.1->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (1.0.0)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.9/site-packages (from requests>=1.25.8->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (3.4)\nRequirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.9/site-packages (from requests>=1.25.8->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (2.0.4)\nRequirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.9/site-packages (from requests>=1.25.8->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (2023.7.22)\nRequirement already satisfied: pycparser in /opt/conda/lib/python3.9/site-packages (from cffi->cbgen>=1.0.4->pysnptools->popcorn==1.0.0) (2.21)\nRequirement already satisfied: platformdirs>=2.5.0 in /opt/conda/lib/python3.9/site-packages (from pooch->cbgen>=1.0.4->pysnptools->popcorn==1.0.0) (3.11.0)\nRequirement already satisfied: zipp>=0.5 in /opt/conda/lib/python3.9/site-packages (from importlib-metadata>=4.13.0->dask[array,bag,dataframe,delayed]>=2.12.0->bgen-reader>=4.0.8->pysnptools->popcorn==1.0.0) (3.17.0)\nCollecting locket\n  Downloading locket-1.0.0-py2.py3-none-any.whl (4.4 kB)\nBuilding wheels for collected packages: popcorn, pysnptools\n  Building wheel for popcorn (setup.py) ... \u001b[?25ldone\n\u001b[?25h  Created wheel for popcorn: filename=popcorn-1.0.0-py3-none-any.whl size=17228 sha256=7afddb621e4df5d9bb0f299513524f593dc25989687f4192c3c0bc8aab4761a1\n  Stored in directory: /tmp/pip-ephem-wheel-cache-vvrj4w2b/wheels/a9/9f/13/e342c997ac040339b4db0e12f9dffa62e71c2f311d26a60145\n  Building wheel for pysnptools (pyproject.toml) ... \u001b[?25ldone\n\u001b[?25h  Created wheel for pysnptools: filename=pysnptools-0.5.11-py3-none-any.whl size=278811 sha256=15735035f764c94d994e99f24ee3438cdcf841d049241d3ed6d33ab9e9af558f\n  Stored in directory: /tmp/pip-ephem-wheel-cache-vvrj4w2b/wheels/d2/d3/1b/910de98922fe9e7f6dbbde255377029bffba19070e5b10fd0f\nSuccessfully built popcorn pysnptools\nInstalling collected packages: texttable, appdirs, patsy, more-itertools, locket, iniconfig, h5py, fsspec, cloudpickle, cachetools, bottleneck, bed-reader, pytest, pooch, partd, xarray, statsmodels, dask, cbgen, bgen-reader, pysnptools, popcorn\nSuccessfully installed appdirs-1.4.4 bed-reader-1.0.0 bgen-reader-4.0.8 bottleneck-1.3.7 cachetools-5.3.2 cbgen-1.0.4 cloudpickle-3.0.0 dask-2023.12.0 fsspec-2023.12.1 h5py-3.10.0 iniconfig-2.0.0 locket-1.0.0 more-itertools-10.1.0 partd-1.4.1 patsy-0.5.4 pooch-1.8.0 popcorn-1.0.0 pysnptools-0.5.11 pytest-7.4.3 statsmodels-0.14.0 texttable-1.7.0 xarray-2023.11.0\n\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n\u001b[0m\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "popcorn -h   # check popcorn was installed correctly", "metadata": {"trusted": true, "tags": []}, "execution_count": 4, "outputs": [{"name": "stdout", "text": "usage: popcorn [-h] {compute,fit} ...\n\npositional arguments:\n  {compute,fit}  Program mode: \"compute\" for computing covariance scores,\n                 \"fit\" for fitting summary statistics with pre-computed\n                 covariance scores.\n\noptional arguments:\n  -h, --help     show this help message and exit\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "# Run Popcorn", "metadata": {}}, {"cell_type": "code", "source": "popcorn fit -h    # shows fit usage", "metadata": {"trusted": true, "tags": []}, "execution_count": 5, "outputs": [{"name": "stdout", "text": "usage: popcorn fit [-h] [-v {0,1,2,3}] [--maf MAF] [--gen_effect] [--h2weight]\n                   [--from_bp FROM_BP] [--to_bp TO_BP] [--sfile SFILE]\n                   [--sfile1 SFILE1] [--sfile2 SFILE2] [--cfile CFILE]\n                   [--Ns NS] [--tol TOL] [--no_jackknife] [--K1 K1] [--P1 P1]\n                   [--K2 K2] [--P2 P2] [--no_intercept] [--plot_likelihood]\n                   [--use_mle]\n                   out\n\npositional arguments:\n  out                Specify output file.\n\noptional arguments:\n  -h, --help         show this help message and exit\n  -v {0,1,2,3}       Specify output verbosity.\n  --maf MAF          Specify MAF cutoff. We suggest at least 0.01.\n  --gen_effect       Compute genetic effect correlation instead of genetic\n                     impact correlation. This must be specified both when\n                     computing scores and fitting the model.\n  --h2weight         Incorporate per-SNP h2 weighting from file. The filename\n                     should be (radix of binary plink file).h2weight\n  --from_bp FROM_BP  Specify base position to start analysis.\n  --to_bp TO_BP      Specify base position to finish analysus.\n  --sfile SFILE      Specify tab delimited summary statistics file\n  --sfile1 SFILE1    Specify tab delimited summary statistics file for\n                     population 1\n  --sfile2 SFILE2    Specify tab delimited summary statistics file for\n                     population 2\n  --cfile CFILE      Specify covariance scores file\n  --Ns NS            Either 1) the number of overlapping samples or 2)\n                     proportion of overlapping samples or a 3) text file\n                     containing the number of overlapping samples per SNP ID.\n  --tol TOL          Specify the convergence tolerance for MLE. Unless you\n                     have convergence issues use default\n  --no_jackknife     Just compute the point estimate without computing the\n                     jackknife standard error.\n  --K1 K1            Specify population prevelance of binary phenotype 1.\n  --P1 P1            Specify study prevelance of binary phenotype 1.\n  --K2 K2            Specify population prevelance of binary phenotype 2.\n  --P2 P2            Specify study prevelance of binary phenotype 2.\n  --no_intercept     Fixes intercept of heritability to be 1.0.\n  --plot_likelihood  Plot the likehood as a function of the parameter.\n  --use_mle          Use original MLE estimator instead of more stable but\n                     higher variance regression estimator.\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "PARENT_DIR=/mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\nSCORES_PREF=${PARENT_DIR}/scores_files\nSUMSTATS_PREF=${PARENT_DIR}/sumstats_files", "metadata": {"trusted": true, "tags": []}, "execution_count": 6, "outputs": []}, {"cell_type": "markdown", "source": "## Overall Breast Cancer Summary Statistics", "metadata": {"tags": []}}, {"cell_type": "markdown", "source": "### African and East Asian populations", "metadata": {}}, {"cell_type": "code", "source": "popcorn fit -v 1 --sfile1 $SUMSTATS_PREF/AFR_meta.sumstats.txt --sfile2 $SUMSTATS_PREF/EAS_meta.sumstats.txt --cfile $SCORES_PREF/AFR_EAS_scores.txt AFR_EAS_meta_corr.txt", "metadata": {"trusted": true}, "execution_count": 29, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob//sumstats_files/AFR_meta.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob//sumstats_files/EAS_meta.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob//scores_files/AFR_EAS_scores.txt AFR_EAS_meta_corr.txt\nBeginning analysis at Mon Nov 20 18:20:48 2023\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n14020290 7533310 SNPs detected in input.\n3785594 SNPs remaining after merging with score file.\n3507462 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.247170 NaN\nh2^2   0.303956 NaN\npg     0.608203 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 17537 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.247170  0.111988  2.207101  2.730701e-02\nh2^2   0.303956  0.053189  5.714636  1.099388e-08\npgi    0.608203  0.264063  1.483722  1.378827e-01\nAnalysis finished at Mon Nov 20 18:41:08 2023\nTotal time elapsed: 1219.65\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "head AFR_EAS_meta_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 30, "outputs": [{"name": "stdout", "text": "\tVal (obs)\tSE\tZ\tP (Z)\nh1^2\t0.24716955653243736\t0.1119883318854628\t2.207100975351901\t0.027307005664357287\nh2^2\t0.3039560217076311\t0.053189039650404765\t5.714636393238922\t1.0993884025367606e-08\npgi\t0.6082034188585443\t0.2640633218985192\t1.4837220797064157\t0.1378826661825463\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "dx upload AFR_EAS_meta_corr.txt*", "metadata": {"trusted": true, "tags": []}, "execution_count": 31, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 271 of 271 bytes (100%) AFR_EAS_meta_corr.txt\nID                          file-GbJpXPQ4BzFzpfjK7F9vBJJj\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_meta_corr.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Nov 20 18:45:18 2023\nCreated by                  omjahagirdar\n via the job                job-GbJk4G04BzFZzjBYJ6KvJbXF\nLast modified               Mon Nov 20 18:45:19 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 383 of 383 bytes (100%) AFR_EAS_meta_corr.txt.e.log\nID                          file-GbJpXQ04BzFZqzf2bGfZVYvf\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_meta_corr.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Nov 20 18:45:20 2023\nCreated by                  omjahagirdar\n via the job                job-GbJk4G04BzFZzjBYJ6KvJbXF\nLast modified               Mon Nov 20 18:45:22 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 12,092 of 12,092 bytes (100%) AFR_EAS_meta_corr.txt.o.log\nID                          file-GbJpXQQ4BzFg540q9qK9VYxV\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EAS_meta_corr.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Nov 20 18:45:22 2023\nCreated by                  omjahagirdar\n via the job                job-GbJk4G04BzFZzjBYJ6KvJbXF\nLast modified               Mon Nov 20 18:45:24 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "### African and European populations", "metadata": {}}, {"cell_type": "code", "source": "popcorn fit -v 1 --sfile1 $SUMSTATS_PREF/AFR_meta.sumstats.txt --sfile2 $SUMSTATS_PREF/EUR_meta.sumstats.txt --cfile $SCORES_PREF/AFR_EUR_scores.txt AFR_EUR_meta_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 32, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob//sumstats_files/AFR_meta.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob//sumstats_files/EUR_meta.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob//scores_files/AFR_EUR_scores.txt AFR_EUR_meta_corr.txt\nBeginning analysis at Mon Nov 20 18:45:39 2023\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n14020290 8452941 SNPs detected in input.\n4114906 SNPs remaining after merging with score file.\n3797861 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.167380 NaN\nh2^2   0.290724 NaN\npg     0.645306 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 18989 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.167380  0.117907  1.419593  1.557263e-01\nh2^2   0.290724  0.046205  6.292002  3.133984e-10\npgi    0.645306  0.277457  1.278374  2.011175e-01\nAnalysis finished at Mon Nov 20 19:06:52 2023\nTotal time elapsed: 1272.64\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "head AFR_EUR_meta_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 33, "outputs": [{"name": "stdout", "text": "\tVal (obs)\tSE\tZ\tP (Z)\nh1^2\t0.1673798482301549\t0.11790694259389517\t1.4195928123304697\t0.15572625916558025\nh2^2\t0.29072420032100926\t0.046205360901084805\t6.292001504833689\t3.133984183278926e-10\npgi\t0.6453064150764202\t0.2774567756189476\t1.2783742048913138\t0.2011175147997174\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "dx upload AFR_EUR_meta_corr.txt*", "metadata": {"trusted": true, "tags": []}, "execution_count": 34, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 271 of 271 bytes (100%) AFR_EUR_meta_corr.txt\nID                          file-GbJpvbQ4BzFb1X6gq62QbqpG\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_corr.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Nov 20 19:07:02 2023\nCreated by                  omjahagirdar\n via the job                job-GbJk4G04BzFZzjBYJ6KvJbXF\nLast modified               Mon Nov 20 19:07:03 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n\nID                          file-GbJpvf04BzFYgzz9pk3Z1jq0\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_corr.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Nov 20 19:07:04 2023\nCreated by                  omjahagirdar\n via the job                job-GbJk4G04BzFZzjBYJ6KvJbXF\nLast modified               Mon Nov 20 19:07:05 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 11,138 of 11,138 bytes (100%) AFR_EUR_meta_corr.txt.o.log\nID                          file-GbJpvfQ4BzFqkBpVvyx7vGKF\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_corr.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Mon Nov 20 19:07:06 2023\nCreated by                  omjahagirdar\n via the job                job-GbJk4G04BzFZzjBYJ6KvJbXF\nLast modified               Mon Nov 20 19:07:08 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "### East Asian and European populations", "metadata": {}}, {"cell_type": "code", "source": "popcorn fit -v 1 --sfile1 $SUMSTATS_PREF/EAS_meta.sumstats.txt --sfile2 $SUMSTATS_PREF/EUR_meta.sumstats.txt --cfile $SCORES_PREF/EAS_EUR_scores.txt EAS_EUR_meta_corr.txt", "metadata": {"trusted": true}, "execution_count": 8, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EAS_meta.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/EAS_EUR_scores.txt EAS_EUR_meta_corr.txt\nBeginning analysis at Fri Dec  1 23:53:49 2023\nAnalyzing a pair of traits in two populations\nTraceback (most recent call last):\n  File \"/opt/conda/bin/popcorn\", line 8, in <module>\n    sys.exit(main())\n  File \"/opt/notebooks/Popcorn/popcorn/__main__.py\", line 197, in main\n    scores = pd.read_table(args.cfile,header=None,comment='#',\n  File \"/opt/conda/lib/python3.9/site-packages/pandas/util/_decorators.py\", line 211, in wrapper\n    return func(*args, **kwargs)\n  File \"/opt/conda/lib/python3.9/site-packages/pandas/util/_decorators.py\", line 331, in wrapper\n    return func(*args, **kwargs)\n  File \"/opt/conda/lib/python3.9/site-packages/pandas/io/parsers/readers.py\", line 1289, in read_table\n    return _read(filepath_or_buffer, kwds)\n  File \"/opt/conda/lib/python3.9/site-packages/pandas/io/parsers/readers.py\", line 611, in _read\n    return parser.read(nrows)\n  File \"/opt/conda/lib/python3.9/site-packages/pandas/io/parsers/readers.py\", line 1778, in read\n    ) = self._engine.read(  # type: ignore[attr-defined]\n  File \"/opt/conda/lib/python3.9/site-packages/pandas/io/parsers/c_parser_wrapper.py\", line 230, in read\n    chunks = self._reader.read_low_memory(nrows)\n  File \"pandas/_libs/parsers.pyx\", line 808, in pandas._libs.parsers.TextReader.read_low_memory\n  File \"pandas/_libs/parsers.pyx\", line 866, in pandas._libs.parsers.TextReader._read_rows\n  File \"pandas/_libs/parsers.pyx\", line 852, in pandas._libs.parsers.TextReader._tokenize_rows\n  File \"pandas/_libs/parsers.pyx\", line 1973, in pandas._libs.parsers.raise_parser_error\npandas.errors.ParserError: Error tokenizing data. C error: Calling read(nbytes) on source failed. Try engine='python'.\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "head EAS_EUR_meta_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 9, "outputs": [{"name": "stdout", "text": "head: cannot open 'EAS_EUR_meta_corr.txt' for reading: No such file or directory\n", "output_type": "stream"}, {"ename": "", "evalue": "1", "traceback": [], "output_type": "error"}]}, {"cell_type": "code", "source": "dx upload EAS_EUR_meta_corr.txt*", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": []}, {"cell_type": "code", "source": "dx ls", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "## ER+/ER- Breast Cancer Summary Statistics", "metadata": {"tags": []}}, {"cell_type": "markdown", "source": "### African and European populations", "metadata": {}}, {"cell_type": "code", "source": "popcorn fit -v 1 --sfile1 $SUMSTATS_PREF/AFR_meta_ERneg.sumstats.txt --sfile2 $SUMSTATS_PREF/EUR_meta_ERneg.sumstats.txt --cfile $SCORES_PREF/AFR_EUR_scores.txt AFR_EUR_meta_ERneg_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_ERneg.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_ERneg.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_EUR_scores.txt AFR_EUR_meta_ERneg_corr.txt\nBeginning analysis at Sat Dec  2 02:45:38 2023\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n14020286 8527375 SNPs detected in input.\n4115319 SNPs remaining after merging with score file.\n3798250 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.034190 NaN\nh2^2   0.329742 NaN\npg     1.203374 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 18991 .\n/opt/notebooks/Popcorn/popcorn/fit.py:288: RuntimeWarning: invalid value encountered in sqrt\n  pg = P_Res(x=res_wls.params.values[1]/np.sqrt(h1.h_res.x[1]*h2.h_res.x[1]))\n/opt/notebooks/Popcorn/popcorn/fit.py:237: RuntimeWarning: invalid value encountered in sqrt\n  hg = res[4].x*np.sqrt(res[0].h_res.x[1]*res[2].h_res.x[1])\n/opt/notebooks/Popcorn/popcorn/fit.py:288: RuntimeWarning: invalid value encountered in sqrt\n  pg = P_Res(x=res_wls.params.values[1]/np.sqrt(h1.h_res.x[1]*h2.h_res.x[1]))\n/opt/notebooks/Popcorn/popcorn/fit.py:237: RuntimeWarning: invalid value encountered in sqrt\n  hg = res[4].x*np.sqrt(res[0].h_res.x[1]*res[2].h_res.x[1])\n/opt/notebooks/Popcorn/popcorn/fit.py:288: RuntimeWarning: invalid value encountered in sqrt\n  pg = P_Res(x=res_wls.params.values[1]/np.sqrt(h1.h_res.x[1]*h2.h_res.x[1]))\n/opt/notebooks/Popcorn/popcorn/fit.py:237: RuntimeWarning: invalid value encountered in sqrt\n  hg = res[4].x*np.sqrt(res[0].h_res.x[1]*res[2].h_res.x[1])\n/opt/notebooks/Popcorn/popcorn/fit.py:288: RuntimeWarning: invalid value encountered in sqrt\n  pg = P_Res(x=res_wls.params.values[1]/np.sqrt(h1.h_res.x[1]*h2.h_res.x[1]))\n/opt/notebooks/Popcorn/popcorn/fit.py:237: RuntimeWarning: invalid value encountered in sqrt\n  hg = res[4].x*np.sqrt(res[0].h_res.x[1]*res[2].h_res.x[1])\n/opt/notebooks/Popcorn/popcorn/fit.py:288: RuntimeWarning: invalid value encountered in sqrt\n  pg = P_Res(x=res_wls.params.values[1]/np.sqrt(h1.h_res.x[1]*h2.h_res.x[1]))\n/opt/notebooks/Popcorn/popcorn/fit.py:237: RuntimeWarning: invalid value encountered in sqrt\n  hg = res[4].x*np.sqrt(res[0].h_res.x[1]*res[2].h_res.x[1])\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.034190  0.240197  0.142342  8.868102e-01\nh2^2   0.329742  0.042811  7.702262  1.332268e-14\npgi    1.203374  9.339853 -0.021775  9.826275e-01\nAnalysis finished at Sat Dec  2 03:07:38 2023\nTotal time elapsed: 1319.75\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "head AFR_EUR_meta_ERneg_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 8, "outputs": [{"name": "stdout", "text": "\tVal (obs)\tSE\tZ\tP (Z)\nh1^2\t0.034190082099148575\t0.24019744985435376\t0.1423415699037608\t0.886810212758585\nh2^2\t0.32974246109057803\t0.042811119486695456\t7.702262053508157\t1.3322676295501878e-14\npgi\t1.2033740628694665\t9.339852710792208\t-0.021774868316121042\t0.9826275416116281\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "dx upload AFR_EUR_meta_ERneg_corr.txt*", "metadata": {"trusted": true, "tags": []}, "execution_count": 9, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 274 of 274 bytes (100%) AFR_EUR_meta_ERneg_corr.txt\nID                          file-Gbb9yyj4BzFpKgfQXf9XxgZ8\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_ERneg_corr.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Sat Dec  2 03:07:39 2023\nCreated by                  omjahagirdar\n via the job                job-Gbb9Vj04BzFqJ65z4FZjYPYj\nLast modified               Sat Dec  2 03:07:40 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 1,625 of 1,625 bytes (100%) AFR_EUR_meta_ERneg_corr.txt.e.log\nID                          file-Gbb9yz84BzFQ6FFjY1GzP4gK\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_ERneg_corr.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Sat Dec  2 03:07:41 2023\nCreated by                  omjahagirdar\n via the job                job-Gbb9Vj04BzFqJ65z4FZjYPYj\nLast modified               Sat Dec  2 03:07:43 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 7,209 of 7,209 bytes (100%) AFR_EUR_meta_ERneg_corr.txt.o.log\nID                          file-Gbb9yzj4BzFgvk57Gy2b4bb0\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_ERneg_corr.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Sat Dec  2 03:07:43 2023\nCreated by                  omjahagirdar\n via the job                job-Gbb9Vj04BzFqJ65z4FZjYPYj\nLast modified               Sat Dec  2 03:07:45 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "popcorn fit -v 1 --sfile1 $SUMSTATS_PREF/AFR_meta_ERpos.sumstats.txt --sfile2 $SUMSTATS_PREF/EUR_meta_ERpos.sumstats.txt --cfile $SCORES_PREF/AFR_EUR_scores.txt AFR_EUR_meta_ERpos_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 19, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob//sumstats_files/AFR_meta_ERpos.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob//sumstats_files/EUR_meta_ERpos.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob//scores_files/AFR_EUR_scores.txt AFR_EUR_meta_ERpos_corr.txt\nBeginning analysis at Tue Nov 28 01:45:56 2023\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n14020279 8527375 SNPs detected in input.\n4115319 SNPs remaining after merging with score file.\n3798250 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.495871 NaN\nh2^2   0.346477 NaN\npg     0.545789 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 18991 .\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.495871  0.203698  2.434344  1.491881e-02\nh2^2   0.346477  0.059439  5.829098  5.572768e-09\npgi    0.545789  0.164834  2.755564  5.859113e-03\nAnalysis finished at Tue Nov 28 02:07:56 2023\nTotal time elapsed: 1320.02\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "head AFR_EUR_meta_ERpos_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 20, "outputs": [{"name": "stdout", "text": "\tVal (obs)\tSE\tZ\tP (Z)\nh1^2\t0.49587060908513647\t0.20369783204383407\t2.4343440679252257\t0.014918806315404876\nh2^2\t0.34647728370449243\t0.05943925939741373\t5.829098262949892\t5.5727683578865594e-09\npgi\t0.5457891123747637\t0.16483412042646253\t2.755563511062465\t0.005859112824758594\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "dx upload AFR_EUR_meta_ERpos_corr.txt*", "metadata": {"trusted": true, "tags": []}, "execution_count": 21, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 275 of 275 bytes (100%) AFR_EUR_meta_ERpos_corr.txt\nID                          file-GbXZfz84BzFQp90J671gQ1Kx\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_ERpos_corr.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Tue Nov 28 02:07:57 2023\nCreated by                  omjahagirdar\n via the job                job-GbXXjyQ4BzFjf8X177yJQvKq\nLast modified               Tue Nov 28 02:07:59 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 2,026 of 2,026 bytes (100%) AFR_EUR_meta_ERpos_corr.txt.e.log\nID                          file-GbXZfzj4BzFkP20YPGYjxP8p\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_ERpos_corr.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Tue Nov 28 02:07:59 2023\nCreated by                  omjahagirdar\n via the job                job-GbXXjyQ4BzFjf8X177yJQvKq\nLast modified               Tue Nov 28 02:08:01 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 12,169 of 12,169 bytes (100%) AFR_EUR_meta_ERpos_corr.txt.o.log\nID                          file-GbXZg0Q4BzFbB3Q4b8228yyB\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_ERpos_corr.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Tue Nov 28 02:08:02 2023\nCreated by                  omjahagirdar\n via the job                job-GbXXjyQ4BzFjf8X177yJQvKq\nLast modified               Tue Nov 28 02:08:03 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "Run the ER+/ER- Popcorn analysis using total sample size rather than effective sample size.", "metadata": {}}, {"cell_type": "code", "source": "popcorn fit -v 1 --sfile1 $SUMSTATS_PREF/AFR_meta_ERneg_totsamples.sumstats.txt --sfile2 $SUMSTATS_PREF/EUR_meta_ERneg_totsamples.sumstats.txt --cfile $SCORES_PREF/AFR_EUR_scores.txt AFR_EUR_meta_ERneg_totsamples_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_ERneg_totsamples.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_ERneg_totsamples.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_EUR_scores.txt AFR_EUR_meta_ERneg_totsamples_corr.txt\nBeginning analysis at Tue Dec  5 19:24:51 2023\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n14020286 8527375 SNPs detected in input.\n4115319 SNPs remaining after merging with score file.\n3798250 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.010224 NaN\nh2^2   0.179006 NaN\npg     0.812597 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 18991 .\n/opt/notebooks/Popcorn/popcorn/fit.py:288: RuntimeWarning: invalid value encountered in sqrt\n  pg = P_Res(x=res_wls.params.values[1]/np.sqrt(h1.h_res.x[1]*h2.h_res.x[1]))\n/opt/notebooks/Popcorn/popcorn/fit.py:237: RuntimeWarning: invalid value encountered in sqrt\n  hg = res[4].x*np.sqrt(res[0].h_res.x[1]*res[2].h_res.x[1])\nJackknife iter: 200 \n      Val (obs)        SE         Z         P (Z)\nh1^2   0.010224  0.036626  0.279151  7.801291e-01\nh2^2   0.179006  0.024619  7.271077  3.566036e-13\npgi    0.812597  1.624682  0.115347  9.081697e-01\nAnalysis finished at Tue Dec  5 19:46:57 2023\nTotal time elapsed: 1326.3\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "head AFR_EUR_meta_ERneg_totsamples_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 8, "outputs": [{"name": "stdout", "text": "\tVal (obs)\tSE\tZ\tP (Z)\nh1^2\t0.010224054518743824\t0.03662555921116179\t0.27915080995208397\t0.780129091563186\nh2^2\t0.17900560688842013\t0.024618858500644835\t7.271076637600053\t3.566036355096003e-13\npgi\t0.8125970592229604\t1.6246815622901265\t0.11534749031858232\t0.9081696977076572\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "dx upload AFR_EUR_meta_ERneg_totsamples_corr.txt*", "metadata": {"trusted": true, "tags": []}, "execution_count": 9, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 273 of 273 bytes (100%) AFR_EUR_meta_ERneg_totsamples_corr.txt\nID                          file-GbggvGQ4BzFq0qB8VF9vx5XV\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_ERneg_totsamples_corr.txt\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Tue Dec  5 19:46:58 2023\nCreated by                  omjahagirdar\n via the job                job-GbggB5Q4BzFgqk7V1GxF5Z42\nLast modified               Tue Dec  5 19:47:00 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 325 of 325 bytes (100%) AFR_EUR_meta_ERneg_totsamples_corr.txt.e.log\nID                          file-GbggvJ84BzFp9J7739k3j2Pf\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_ERneg_totsamples_corr.txt.e.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Tue Dec  5 19:47:01 2023\nCreated by                  omjahagirdar\n via the job                job-GbggB5Q4BzFgqk7V1GxF5Z42\nLast modified               Tue Dec  5 19:47:02 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 5,941 of 5,941 bytes (100%) AFR_EUR_meta_ERneg_totsamples_corr.txt.o.log\nID                          file-GbggvJj4BzFQjQjfk11P5Fp6\nClass                       file\nProject                     project-GYBZ3yj4BzFgB61K7vX9Fq9K\nFolder                      /Aim4_CrossAncestryGeneticCorrelation/jahagirdarob\n                            /corr_files\nName                        AFR_EUR_meta_ERneg_totsamples_corr.txt.o.log\nState                       \u001b[33mclosing\u001b[0m\nVisibility                  visible\nTypes                       -\nProperties                  -\nTags                        -\nOutgoing links              -\nCreated                     Tue Dec  5 19:47:03 2023\nCreated by                  omjahagirdar\n via the job                job-GbggB5Q4BzFgqk7V1GxF5Z42\nLast modified               Tue Dec  5 19:47:05 2023\nMedia type                  \narchivalState               \"live\"\ncloudAccount                \"cloudaccount-dnanexus\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "popcorn fit -v 1 --sfile1 $SUMSTATS_PREF/AFR_meta_ERpos_totsamples.sumstats.txt --sfile2 $SUMSTATS_PREF/EUR_meta_ERpos_totsamples.sumstats.txt --cfile $SCORES_PREF/AFR_EUR_scores.txt AFR_EUR_meta_ERpos_totsamples_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [{"name": "stdout", "text": "Popcorn version 1.0\n(C) 2015-2019 Brielin C Brown\nUniversity of California, Berkeley\nGNU General Public License v3\n\nInvoking command: python /opt/conda/bin/popcorn fit -v 1 --sfile1 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/AFR_meta_ERpos_totsamples.sumstats.txt --sfile2 /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/sumstats_files/EUR_meta_ERpos_totsamples.sumstats.txt --cfile /mnt/project/Aim4_CrossAncestryGeneticCorrelation/jahagirdarob/scores_files/AFR_EUR_scores.txt AFR_EUR_meta_ERpos_totsamples_corr.txt\nBeginning analysis at Tue Dec  5 19:47:06 2023\nAnalyzing a pair of traits in two populations\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\nNote: Study AF not provided. Defulating to ignoring variants where minor allele is the compliment of the major allele.\n14020279 8527375 SNPs detected in input.\n4115319 SNPs remaining after merging with score file.\n3798250 SNPs remaining after filtering on AF and self-compliment SNPs.\nInitial estimate:\n       Val (obs)  SE\nh1^2   0.099191 NaN\nh2^2   0.149176 NaN\npg     0.524530 NaN\nPerforming jackknife estimation of the standard error using 200 blocks of size 18991 .\nJackknife iter: 192 \r", "output_type": "stream"}]}, {"cell_type": "code", "source": "head AFR_EUR_meta_ERpos_totsamples_corr.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": []}, {"cell_type": "code", "source": "dx upload AFR_EUR_meta_ERpos_totsamples_corr.txt*", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": []}, {"cell_type": "code", "source": "dx ls", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": []}]}